### PR TITLE
New feature: a simple implement of voice promt

### PR DESCRIPTION
**Pull request by jijiqw**
Add an option "voice promt" in PreferencesManager , it play an "ding dong" sound  when you change to active player. just like the notification  "It's your turn!"

### CODE REVIEW

File: src/PlaySound.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    oscillator.frequency.value = frequency;
+    gainNode.gain.setValueAtTime(0, time);
+    gainNode.gain.linearRampToValueAtTime(1, time + 0.01);
+    oscillator.start(time);
+    gainNode.gain.exponentialRampToValueAtTime(0.001, time + len);
+    oscillator.stop(time + len);
+};
+
+export function voiceForActivePlayer(){
+    if(window.AudioContext){
```
Comment (nwai90) [inline src/PlaySound.ts:33]: Personal preference: Shorter sound e..g

```
if (window.AudioContext){
        playSound(frequency(10),0,0.4);
        playSound(frequency(8),0.2,0.4);
    }
```

File: src/PlaySound.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+window.AudioContext = window.AudioContext || (window as any).webkitAudioContext;
```
Comment (nwai90) [inline src/PlaySound.ts:1]: This file may be better named as `Sounds.ts` or `SoundManager.ts`

File: src/PlaySound.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+if (!window.AudioContext) { 
+    console.log("This web browser doesn't support Web Audio API"); 
+}
+
+
+let audioCtx = window.AudioContext ?  new AudioContext() : undefined;
+let arrFrequency = [196.00, 220.00, 246.94, 261.63, 293.66, 329.63, 349.23, 392.00, 440.00, 493.88, 523.25, 587.33, 659.25, 698.46, 783.99, 880.00, 987.77, 1046.50];
```
Comment (nwai90) [inline src/PlaySound.ts:8]: This may be better as a constant

File: src/PlaySound.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    return  arrFrequency[index];
+}
+
+function playSound (frequency: number , time: number , len: number) {
+    time = audioCtx!.currentTime + time;
+    let  oscillator = audioCtx!.createOscillator();
+    let  gainNode = audioCtx!.createGain();
+    oscillator.connect(gainNode);
+    gainNode.connect(audioCtx!.destination);
+    oscillator.type = "sine";
```
Comment (nwai90) [inline src/PlaySound.ts:23]: Minor: This function could be better organised / split up as it's doing three things:

1) Configuring oscillator
2) Configuring gainNode
3) Playing the sound

**Comment by nwai90** [comment]
> Add an option "voice promt" in PreferencesManager , it play an "ding dong" sound when you change to active player. just like the notification "It's your turn!"

Nice addition!

**Comment by jijiqw** [comment]
Be free to  refactor youself , that might be more  convenient and efficiently IMO  :-D

**Comment by jijiqw** [comment]
image

**Comment by chosta** [comment]
Great feature but from my experience sound could cause issues in different browsers. Did you test it in the most used ones:
chrome, zilla, edge, safari ?
also on mobile?

**Comment by jijiqw** [comment]
> Great feature but from my experience sound could cause issues in different browsers. Did you test it in the most used ones:
> chrome, zilla, edge, safari ?
> also on mobile?

Nope

**Comment by nwai90** [comment]
Will reopen this and work on it at some point when have more time. It's a great feature to have; starting with just 1 sound would be nice

**Pull request closed without merging**